metricbeat/docs/modules/sql.asciidoc

```diff
@@ -8,9 +8,361 @@ This file is generated! See scripts/mage/docs_collector.go
+
+  driver: "oracle"
+  sql_query: 'SELECT name, physical_reads, db_block_gets, consistent_gets, 1 - (physical_reads / (db_block_gets + consistent_gets)) "Hit Ratio" FROM V$BUFFER_POOL_STATISTICS'
+  sql_response_format: table
+----
+
+
+[source,json]
+----
+{
+  "@timestamp": "2020-06-09T15:41:02.200Z",
+  "@metadata": {
+    "beat": "metricbeat",
+    "type": "_doc",
+    "version": "8.0.0"
+  },
+  "sql": {
+    "metrics": {
+      "numeric": {
+        "hit ratio": 0.9742963357937117,
+        "physical_reads": 17161,
+        "db_block_gets": 122221,
+        "consistent_gets": 545427
+      },
+      "string": {
+        "name": "DEFAULT"
+      }
+    },
+    "driver": "oracle",
+    "query": "SELECT name, physical_reads, db_block_gets, consistent_gets, 1 - (physical_reads / (db_block_gets + consistent_gets)) \"Hit Ratio\" FROM V$BUFFER_POOL_STATISTICS"
+  },
+  "metricset": {
+    "period": 10000,
+    "name": "query"
+  },
+  "service": {
+    "address": "172.17.0.3:1521",
+    "type": "sql"
+  },
+  "event": {
+    "dataset": "sql.query",
+    "module": "sql",
+    "duration": 39233704
+  },
+  "ecs": {
+    "version": "1.5.0"
+  },
+  "host": {
+    "name": "elastic"
+  },
+  "agent": {
+    "id": "670ef211-87f0-4f38-8beb-655c377f1629",
+    "name": "elastic",
+    "type": "metricbeat",
+    "version": "8.0.0",
+    "ephemeral_id": "49e00060-0fa4-4b34-80f1-446881f7a788"
+  }
+}
+----
+
+=== MSSQL
+
+Get the buffer cache hit ratio:
+
+.sql.yml
+[source,yaml]
+----
+- module: sql
+  metricsets:
+    - query
+  period: 10s
+  hosts: ["sqlserver://SA:password@localhost"]
+
+  driver: "mssql"
+  sql_query: 'SELECT * FROM sys.dm_db_log_space_usage'
+  sql_response_format: table
+----
+
+[source,json]
+----
+{
+  "@timestamp": "2020-06-09T15:39:14.421Z",
+  "@metadata": {
+    "beat": "metricbeat",
+    "type": "_doc",
+    "version": "8.0.0"
+  },
+  "sql": {
+    "driver": "mssql",
+    "query": "SELECT * FROM sys.dm_db_log_space_usage",
+    "metrics": {
+      "numeric": {
+        "log_space_in_bytes_since_last_backup": 524288,
+        "database_id": 1,
+        "total_log_size_in_bytes": 2.08896e+06,
+        "used_log_space_in_bytes": 954368,
+        "used_log_space_in_percent": 45.686275482177734
+      }
+    }
+  },
+  "event": {
+    "dataset": "sql.query",
+    "module": "sql",
+    "duration": 40750570
+  },
+  "metricset": {
+    "name": "query",
+    "period": 10000
+  },
+  "service": {
+    "address": "172.17.0.2",
+    "type": "sql"
+  },
+  "agent": {
+    "id": "670ef211-87f0-4f38-8beb-655c377f1629",
+    "name": "elastic",
+    "type": "metricbeat",
+    "version": "8.0.0",
+    "ephemeral_id": "3da88889-036e-47cb-a88b-275037fa2bc9"
+  },
+  "ecs": {
+    "version": "1.5.0"
+  },
+  "host": {
+    "name": "elastic"
+  }
+}
+----
+
+=== Two or more queries
+
+If you want to launch two or more queries, you need to specify them with their full configuration for each query. For example:
```

Comment:
Not sure if it would be more convenient to do something similar to Prometheus Query metricset where we define multiple queries within a single module config. see https://github.com/elastic/beats/blob/master/metricbeat/modules.d/prometheus.yml.disabled#L37

Reply:
I see your point and, in fact, it is implemented that way in the MySQL module (also in this PR) but I'd prefer to not add more complexity to the PR (it's already very big) and leave that change for a future discussion. The reason is because I feel like it's not a very good idea to make the `sql` module very big and / or "intelligent" and rather give more power to `query` metricsets in the rest of the modules, which can handle more things like the DSNs, auth, etc.

How does it sounds?

Reply:
Fair enough! Thanks for elaborating!